### PR TITLE
[PECO-338] Generate and upload new Thrift definitions

### PR DESCRIPTION
**Pull request by mattdeekay**
### Description
Upgrade to Thrift 0.17 and use newest DBR Thrift protocol generated definitions

### Open questions
- [ ] Should we remove `cli_service.thrift`?

### CODE REVIEW

**Review by susodapop — APPROVED** [review]
LGTM. Tests pass. This is a great step towards GA 🚀 

> Should we remove cli_service.thrift?

Yes. `cli_service.thrift` was copied from the official HiveServer implementation. This repository does not need it from here on.

File: cli_service/cli_service-consts.go [hunk 1]
```diff
@@ -1,21 +1,22 @@
-// Autogenerated by Thrift Compiler (0.12.0)
-// DO NOT EDIT UNLESS YOU ARE SURE THAT YOU KNOW WHAT YOU ARE DOING
+// Code generated by Thrift Compiler (0.17.0). DO NOT EDIT.
```
Comment (moderakh) [inline cli_service/cli_service-consts.go:1]: @mattdeekay why none of the databricks thrift definition optimization (e.g., directResult) is in this PR? 

did we use databricks thrift protocol to update these files or did we use the newer open source thrift protocol to update the thrift definition in this PR?

Comment (susodapop) [reply cli_service/cli_service-consts.go:1]: @moderakh the Databricks specific optimisations _are present_ in this updated file. But this PR doesn't include any of the implementations of those interfaces. This PR represents the "lift" portion of "lift and shift" from the OSS Apache Hive Thrift definition file.

> did we use databricks thrift protocol to update these files or did we use the newer open source thrift protocol to update the thrift definition in this PR?

@mattdeekay generated these new stubs using the private Databricks Thrift definition.

**Review by moderakh — APPROVED** [review]
Thanks @mattdeekay and @susodapop great work.

File: cli_service/cli_service-consts.go (hunk 1 above)
Comment (moderakh) [reply cli_service/cli_service-consts.go:1]: Screen Shot 2022-09-22 at 1 49 41 PM
I was looking for the changes to `cli_service.go` expecting that to happen in this PR. but github was not showing any changes in that file. (see the screenshot attached). I can see the changes now by going to the specific commit.

**Pull request merged**